frontend/components/commons/dataset-train/DatasetTrain.component.vue

```diff
@@ -0,0 +1,205 @@
+  </div>
+</template>
+
+<script>
+export default {
+  props: {
+    datasetTask: {
+      type: String,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      selectedComponent: null,
+      currentTaskLibraries: [],
+    };
+  },
+  async fetch() {
+    const libraries = this.getLibraries();
+    this.getCurrentTaskLibraries(libraries).forEach(async (library) => {
+      const newLib = await libraries(library);
+      this.currentTaskLibraries.push(newLib);
+    });
+  },
+  computed: {
+    sortedSnippetsTabs() {
+      return this.snippetsTabs.sort();
+    },
+    visibleTab() {
+      return this.selectedComponent || this.snippetsTabs[0] || null;
+    },
+    snippetsTabs() {
+      return this.currentTaskLibraries.map((library) => {
+        return this.getLibrary(library?.attributes?.title);
+      });
+    },
+    task() {
+      switch (this.datasetTask) {
+        case "TextClassification":
+          return "text-classification";
+        case "TokenClassification":
+          return "token-classification";
+        case "Text2Text":
+          return "text2text";
+      }
+    },
+    parsedSnippet() {
+      const docElement = new DOMParser().parseFromString(
+        this.snippet.html,
+        "text/html"
+      ).documentElement;
+      const preBlocks = docElement.getElementsByTagName("pre");
+      for (let i = 0; i < preBlocks.length; i++) {
+        const code = preBlocks[i].innerText;
+        preBlocks[i].innerHTML = `<base-code code='${code}'></base-code>`;
+      }
+      const html = docElement.getElementsByTagName("body")[0].innerHTML;
+      const htmlWithVariables = html
+        .replace("<my_dataset_name>", this.datasetName)
+        .replace("<my_workspace_name>", this.workspaceName);
+      return `<div>${htmlWithVariables}</div>`;
+    },
+    snippetAttributes() {
+      return this.snippet?.attributes;
+    },
+    snippet() {
+      return (
+        this.currentTaskLibraries.find(
+          (library) => library.attributes.title === this.visibleTab.name
+        ) || {}
+      );
+    },
+    workspaceName() {
+      return this.$route.params.workspace;
+    },
+    datasetName() {
+      return this.$route.params.dataset;
+    },
+  },
+  methods: {
+    getCurrentTaskLibraries(libraries) {
+      return (
+        libraries?.keys().filter((library) => library.includes(this.task)) || []
+      );
+    },
+    getSelectedLibrary(id) {
+      this.selectedComponent = this.snippetsTabs.find(
+        (snippet) => snippet.id === id
+      );
+    },
+    getLibrary(name) {
+      return {
+        id: name.trim().toLowerCase(),
+        name,
+      };
+    },
+    getLibraries() {
+      let libraries = null;
+      try {
+        libraries = require.context(
+          `../../../../docs/_source/_common/snippets/training`,
+          true,
+          /^[^_]+\.md$/,
+          "lazy"
+        );
+      } catch (e) {
+        console.log(e);
+      }
+      return libraries;
+    },
+  },
+};
```

Comment:
performance: dont't computed variable which don't need reactivity. This component have a lot of computeds. Maybe some or all  can be put in static variables => see EditionUserInfo created hook. (if some value are not well reactive, you could use a conditional rendering at the level parent with a v-if)